karapace/serialization.py

```diff
@@ -0,0 +1,166 @@
+from avro.io import BinaryDecoder, BinaryEncoder, DatumReader, DatumWriter
+from karapace.config import read_config
+from karapace.utils import json_encode
+from tests.utils import Client
+from urllib.parse import quote
+
+import aiohttp
+import avro
+import io
+import logging
+import struct
+
+log = logging.getLogger(__name__)
+
+START_BYTE = 0x13
+HEADER_FORMAT = ">bI"
+HEADER_SIZE = 5
+
+
+class InvalidMessageHeader(Exception):
+    pass
+
+
+class InvalidPayload(Exception):
+    pass
+
+
+class InvalidMessageSchema(Exception):
+    pass
+
+
+class SchemaError(Exception):
+    pass
+
+
+class SchemaRetrievalError(SchemaError):
+    pass
+
+
+class SchemaUpdateError(SchemaError):
+    pass
+
+
+def topic_name_strategy(topic_name, record_name):
+    # pylint: disable=W0613
+    return topic_name
+
+
+def record_name_strategy(topic_name, record_name):
+    # pylint: disable=W0613
+    return record_name
+
+
+def topic_record_name_strategy(topic_name, record_name):
+    return topic_name + "-" + record_name
+
+
+NAME_STRATEGIES = {
+    "topic_name": topic_name_strategy,
+    "record_name": record_name_strategy,
+    "topic_record_name": topic_record_name_strategy
+}
+
+
+class SchemaRegistryClient:
+    def __init__(self, schema_registry_url: str = "http://localhost:8081", loop=None):
+        self.client = Client(server_uri=schema_registry_url, client=aiohttp.ClientSession(loop=loop))
+        self.base_url = schema_registry_url
+
+    async def post_new_schema(self, subject, schema):
+        payload = {"schema": json_encode(schema.to_json())}
+        result = await self.client.post("subjects/%s/versions" % quote(subject), json=payload)
+        if result.status_code != 200:
+            raise SchemaRetrievalError(result.json())
+        return result.json()["id"]
+
+    async def get_latest_schema(self, subject):
+        result = await self.client.get("subjects/%s/versions/latest" % quote(subject))
+        if result.status_code != 200:
+            raise SchemaRetrievalError(result.json())
```

Comment:
This bit seems to be repeated.

Reply:
it's repeated once for the normal `json()` method and then appears once more for the `message` member of the json body, which is only there __if__ the status code is not OK. Pulling this into a separate method for one repeat does not make much sense to me